Section safe subgraph generation tool (#249)
graphprotocol/docs: pages/en/developing/creating-a-subgraph.mdx

### Built-In Scalar Types

> 
> 
> 
> 
> GraphQL Supported Scalars

Reviewer: ```suggestion
#### GraphQL Supported Scalars
```

Code generation does not check your mapping code in `src/mapping.ts`. If you want to check that before trying to deploy your subgraph to the Graph Explorer, you can run `yarn build` and fix any syntax errors that the TypeScript compiler might find.

### Safe Subgraph Code Generator

> [Subgraph Uncrashable](https://float-capital.github.io/float-subgraph-uncrashable/) is code generation tool that generates a set of helper functions from the graphql schema of the project and  ensures that all interactions with entities in the your subgraph are completely safe and consistent.

Reviewer 2: ```suggestion
[Subgraph Uncrashable](https://float-capital.github.io/float-subgraph-uncrashable/) is a code generation tool that generates a set of helper functions from the graphql schema of the project and  ensures that all interactions with entities in the your subgraph are completely safe and consistent.
```

#### Why integrate with Subgraph Uncrashable?

**Continuous Uptime**

> Mishandled entities cause subgraphs to crash, which can be very disruptive for projects that are dependent on the graph. Set up helper functions to make your subgraphs “uncrashable” and ensure business continuity.

Reviewer 2: ```suggestion
Mishandled entities cause subgraphs to crash, which can be very disruptive for projects that are dependent on The Graph. Set up helper functions to make your subgraphs “uncrashable” and ensure business continuity.
```

**Key Features**

The code generation tool accommodates **all** subgraph types and is configurable for users to set sane defaults on values. The code generation will use this config to generate helper functions that are to the users specification.

The framework also includes a way (via the config file) to create custom but safe setter functions for groups of entity variables. This way it is impossible for the user to load/use a stale graph entity and it is also impossible to forget to save or set a variable that is required by the function.

Warning logs are recorded as logs indicating where there is a breach of subgraph logic to help patch the issue to ensure data accuracy. These logs can be viewed in the The Graph's hosted service under the 'Logs' section.

Subgraph Uncrashable can be run as an optional flag using the Graph CLI codegen command.

```sh
graph codegen [--output-dir <OUTPUT_DIR>] [<MANIFEST>]
```

Run the graph uncrashable codegen via the graph-cli:

```sh

> graph codegen -u`

Reviewer 2: ```suggestion
graph codegen -u
```

```

> Visit the [subgraph uncrashable documentation](https://float-capital.github.io/float-subgraph-uncrashable/docs/) or watch this [video tutorial](https://float-capital.github.io/float-subgraph-uncrashable/docs/tutorial) to learn more and to get started with developing safer subgraphs. 

Reviewer 3: @SvenMuller95 it seems like the video is no longer available. Will this be a new one demonstrating how it may be used using graph-cli instead of the standalone tool like you had before?

Reviewer 3: Oh, just saw your e-mail... It seems like you're still in the process of updating this link with the new tutorial, right? I now see the new video @ https://www.youtube.com/watch?v=6FaC44njxzU

Author: Hi Pedro, correct. We're doing a dev to main merge today , after which the video will be replaced. We had deleted the old one and recorded a new one with graph-cli.

**User Configurable**

Set default values and configure the level of security checks that suits your individual project's needs. Warning logs are recorded indicating where there is a breach of subgraph logic to help patch the issue to ensure data accuracy.

**Key Features**

The code generation tool accommodates **all** subgraph types and is configurable for users to set sane defaults on values. The code generation will use this config to generate helper functions that are to the users specification.

The framework also includes a way (via the config file) to create custom but safe setter functions for groups of entity variables. This way it is impossible for the user to load/use a stale graph entity and it is also impossible to forget to save or set a variable that is required by the function.

Warning logs are recorded as logs indicating where there is a breach of subgraph logic to help patch the issue to ensure data accuracy. These logs can be viewed in the The Graph's hosted service under the 'Logs' section.

> Subgraph Uncrashable can be installed using the Graph CLI as an optional flag.

Reviewer: ```suggestion
Subgraph Uncrashable can be run as an optional flag using the Graph CLI codegen command.
```
